tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/otextension/BristolRotBatch.java

```diff
@@ -42,72 +37,51 @@ public BristolRotBatch(RotFactory randomOtExtension, int comSecParam, int statSe
-        numMessages);
-    int amountToPreprocess = computeExtensionSize(numMessages,
-        comSecParam, statSecParam);
-    Pair<List<StrictBitVector>, List<StrictBitVector>> messages = sender
-        .extend(amountToPreprocess);
-    List<StrictBitVector> rawZeroMessages = messages.getFirst();
-    List<StrictBitVector> rawOneMessages = messages.getSecond();
-    for (int i = 0; i < numMessages; i++) {
-      StrictBitVector zeroMessage = computeRandomMessage(rawZeroMessages.get(i),
-          sizeOfEachMessage);
-      StrictBitVector oneMessage = computeRandomMessage(rawOneMessages.get(i),
-          sizeOfEachMessage);
-      Pair<StrictBitVector, StrictBitVector> currentPair = new Pair<>(
-          zeroMessage, oneMessage);
-      res.add(currentPair);
-    }
-    return res;
+    int amountToPreprocess = computeExtensionSize(numMessages, comSecParam, statSecParam);
+    Pair<List<StrictBitVector>, List<StrictBitVector>> messages = sender.extend(amountToPreprocess);
+    List<StrictBitVector> zeroMessages = messages.getFirst().stream().limit(numMessages)
+        .map(m -> LengthAdjustment.adjust(m.toByteArray(), sizeOfEachMessage / Byte.SIZE))
+        .map(StrictBitVector::new)
+        .collect(Collectors.toList());
+    List<StrictBitVector> oneMessages = messages.getSecond().stream().limit(numMessages)
+        .map(m -> LengthAdjustment.adjust(m.toByteArray(), sizeOfEachMessage / Byte.SIZE))
+        .map(StrictBitVector::new)
+        .collect(Collectors.toList());
+    return IntStream.range(0, numMessages)
+        .mapToObj(i -> new Pair<>(zeroMessages.get(i), oneMessages.get(i)))
+        .collect(Collectors.toList());
   }
 
   @Override
-  public List<StrictBitVector> receive(StrictBitVector choiceBits,
-      int sizeOfEachMessage) {
+  public List<StrictBitVector> receive(StrictBitVector choiceBits, int sizeOfEachMessage) {
     if (this.receiver == null) {
       this.receiver = rot.createReceiver();
     }
-    List<StrictBitVector> res = new ArrayList<>(choiceBits.getSize());
-    // Find how many OTs we need to preprocess
-    int amountToPreprocess = computeExtensionSize(choiceBits.getSize(),
-        comSecParam, statSecParam);
-    // Construct a new choice-bit vector of the original choices, padded with
-    // 0 choices if needed
-    byte[] extraByteChoices = Arrays.copyOf(choiceBits.toByteArray(),
-        amountToPreprocess / 8);
-    StrictBitVector extraChoices = new StrictBitVector(extraByteChoices);
-    List<StrictBitVector> messages = receiver.extend(extraChoices);
-    for (int i = 0; i < choiceBits.getSize(); i++) {
-      StrictBitVector newMessage = computeRandomMessage(messages.get(i),
-          sizeOfEachMessage);
-      res.add(newMessage);
-    }
-    return res;
+    int amountToPreprocess = computeExtensionSize(choiceBits.getSize(), comSecParam, statSecParam);
+    byte[] extraByteChoices = Arrays.copyOf(choiceBits.toByteArray(), amountToPreprocess / 8);
```

Comment:
Shouldn't it be  Byte.SIZE instead of 8, just to be consistent in the usage of constants :)